lib/action_cable_client/message_factory.rb

```diff
@@ -1,11 +1,16 @@
 # frozen_string_literal: true
 class ActionCableClient
   class MessageFactory
-    attr_reader :_channel
+    attr_reader :identifier
 
-    # @param [String] channel - the name of the subscribed channel
+    # @param [String or Hash] channel - the name of the subscribed channel, or
+    #   a hash that includes the :channel key and any other params to send.
     def initialize(channel)
-      @_channel = channel
```

Comment:
for what it's worth, I'm implementing this with some tests, and will push a PR soon. I'll give you first credit in the changelog